File: 321/321C.cpp
#include <iostream>
using namespace std;

bool is_321_like(int num) {
    string str_num = to_string(num);
    for (size_t i = 0; i < str_num.length() - 1; i++) {
        if (str_num[i] <= str_num[i + 1]) {
            return false;
        }
    }
    return true;
}

int find_kth_321_like_number(int K) {
    int current_num = 1;
    while (K > 0) {
        if (is_321_like(current_num)) {
            K--;
        }
        current_num++;
    }
    return current_num - 1;
}

int main() {
    int K;
    cin >> K;
    int result = find_kth_321_like_number(K);
    cout << result << endl;
    return 0;
}

// AC 28 TLE 5

File: 321/321C.py
def find_kth_321_like_number(K):
    current_number = 1
    kth_number = None
    while K > 0:
        kth_number = current_number
        current_number += 1
        str_num = str(kth_number)
        is_321_like = True
        for i in range(len(str_num)-1):
            if str_num[i] <= str_num[i+1]:
                is_321_like = False
                break
        if is_321_like:
            K -= 1
    return kth_number

K = int(input())
result = find_kth_321_like_number(K)
print(result)
